source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -33,17 +33,38 @@ FileImplWin32::~FileImplWin32() {
   }
 }
 
-void FileImplWin32::openFile() {
-  const int flags = _O_RDWR | _O_APPEND | _O_CREAT;
-  const int mode = _S_IREAD | _S_IWRITE;
-
-  fd_ = ::_open(path_.c_str(), flags, mode);
+void FileImplWin32::openFile(FlagSet in) {
+  const auto flags_and_perm = translateFlag(in);
+  fd_ = ::open(path_.c_str(), flags_and_perm.flags, flags_and_perm.perm);
```

Comment:
ah I finally figured out what 'perm' meant by reading the MS docs: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/open-wopen?view=vs-2019 -- it means "permissions".

but there it's called `pmode`, not `perm`.

Reply:
Thanks and will change